src/utils/CCTPUtils.ts

```diff
@@ -17,6 +19,107 @@ export type DecodedCCTPMessage = {
+ * @param sourceSearchConfig The search configuration to use when querying the sourceTokenMessenger contract via `paginatedEventQuery`.
+ * @param sourceToken The token address of the token being transferred.
+ * @param sourceChainId The chainId of the source chain.
+ * @param destinationChainId The chainId of the destination chain.
+ * @param fromAddress The address that initiated the transfer.
+ * @returns A list of outstanding CCTP bridge transfers. These are transfers that have been initiated but not yet finalized on the destination chain.
+ * @dev Reference `hasCCTPMessageBeenProcessed` for more information on how the message is determined to be processed.
+ */
+export async function retrieveOutstandingCCTPBridgeUSDCTransfers(
+  sourceTokenMessenger: ethers.Contract,
+  destinationMessageTransmitter: ethers.Contract,
+  sourceSearchConfig: EventSearchConfig,
+  sourceToken: string,
+  sourceChainId: number,
+  destinationChainId: number,
+  fromAddress: string
+): Promise<ethers.Event[]> {
+  const sourceDomain = chainIdsToCctpDomains[sourceChainId];
+  const targetDestinationDomain = chainIdsToCctpDomains[destinationChainId];
+
+  const sourceFilter = sourceTokenMessenger.filters.DepositForBurn(undefined, sourceToken, undefined, fromAddress);
+  const initializationTransactions = await paginatedEventQuery(sourceTokenMessenger, sourceFilter, sourceSearchConfig);
+
+  const outstandingTransactions = await Promise.all(
+    initializationTransactions.map(async (event) => {
+      const { nonce, destinationDomain } = event.args;
+      // Ensure that the destination domain matches the target destination domain so that we don't
+      // have any double counting of messages.
+      if (destinationDomain !== targetDestinationDomain) {
+        return undefined;
+      }
+      // Call into the destinationMessageTransmitter contract to determine if the message has been processed
+      // on the destionation chain. We want to make sure the message **hasn't** been processed.
+      const isMessageProcessed = await hasCCTPMessageBeenProcessed(sourceDomain, nonce, destinationMessageTransmitter);
+      if (isMessageProcessed) {
+        return undefined;
+      }
+      return event;
+    })
+  );
+
+  return outstandingTransactions.filter(isDefined);
+}
+
+/**
+ * Calls into the CCTP MessageTransmitter contract and determines whether or not a message has been processed.
+ * @param sourceDomain The source domain of the message.
+ * @param nonce The nonce of the message.
+ * @param contract The CCTP MessageTransmitter contract to call.
+ * @returns Whether or not the message has been processed.
+ */
+export async function hasCCTPMessageBeenProcessed(
+  sourceDomain: number,
+  nonce: number,
+  contract: ethers.Contract
+): Promise<boolean> {
+  const nonceHash = hashCCTPSourceAndNonce(sourceDomain, nonce);
+  const resultingCall = (await contract.callStatic.usedNonces(nonceHash)) as BigNumber;
```

Comment:
would this be better or is it the same?
```suggestion
  const resultingCall: BigNumber = await contract.callStatic.usedNonces(nonceHash);
```

Reply:
It's not a huge difference. One is an explicit type cast of the variable and the other is an implicit typecast of the variable based on the assignment. I'll change this to `resultingCall: BigNumber` instead.